api/app_analytics/cache.py

```diff
@@ -31,3 +33,51 @@ def track_request(self, resource: int, host: str, environment_key: str):
             self._cache[key] += 1
         if (timezone.now() - self._last_flushed_at).seconds > CACHE_FLUSH_INTERVAL:
             self._flush()
+
+
+class FeatureEvaluationCache:
+    def __init__(self):
+        self._cache = {}
+        self._last_flushed_at = timezone.now()
+
+    def _flush(self):
+        evaluation_data = {}
+        for (environment_id, feature_name), eval_count in self._cache.items():
+            if environment_id in evaluation_data:
+                evaluation_data[environment_id][feature_name] = eval_count
+            else:
+                evaluation_data[environment_id] = {feature_name: eval_count}
+
+        for environment_id, feature_evaluations in evaluation_data.items():
+            if settings.USE_POSTGRES_FOR_ANALYTICS:
+                track_feature_evaluation.delay(
+                    kwargs={
+                        "environment_id": environment_id,
+                        "feature_evaluations": feature_evaluations,
+                    }
+                )
+
+            elif settings.INFLUXDB_TOKEN:
+                track_feature_evaluation_influxdb.delay(
+                    kwargs={
+                        "environment_id": environment_id,
+                        "feature_evaluations": feature_evaluations,
+                    }
+                )
+
+        self._cache = {}
+        self._last_flushed_at = timezone.now()
+
+    def track_feature_evaluation(
+        self, environment_id: int, feature_name: str, evaluation_count: int
+    ):
+        key = (environment_id, feature_name)
+        if key not in self._cache:
+            self._cache[key] = evaluation_count
+        else:
+            self._cache[key] += evaluation_count
+
+        if (
+            timezone.now() - self._last_flushed_at
+        ).seconds > settings.FEATURE_EVALUATION_CACHE_SECONDS:
+            self._flush()
```

Comment:
Just to clarify: since the cache is in-memory by default, we have to be ready to lose unsent evaluations.

If it's not an option, we probably should add a destructor that calls the `_flush` method, or come up with a better solution that guarantees the flush in case the app is shut down with a populated cache.

Reply:
I don't think we are okay with that @matthewelwell ?

Reply:
Yeah, these figures don't need to be super accurate. I think that's fine.